Lombiq.NodeJs.Extensions.Samples/Lombiq.NodeJs.Extensions.Samples.csproj

```diff
@@ -9,6 +9,7 @@
     <NodeJsExtensionsScriptsTargetFolder>wwwCustomRoot/jay-es</NodeJsExtensionsScriptsTargetFolder>
     <NodeJsExtensionsStylesSourceFolder>NonDefaultScssFolder</NodeJsExtensionsStylesSourceFolder>
     <NodeJsExtensionsStylesTargetFolder>wwwwNonDefaultCssFolder</NodeJsExtensionsStylesTargetFolder>
+    <MarkdownAnalysis>solution</MarkdownAnalysis>
```

Comment:
Shouldn't we rather have this in the OSOCE Directory.Build.props?

Reply:
No, because you only want to do it once, not for every project. Same logic as with PowerShell analysis.

Reply:
And is there nothing for this in a props, like "add this once for the solution"? It would be strange to add such solution-wide config to one particular project in each solution.

Reply:
I was thinking the same thing, and I'm still wondering on how to approach that more cleanly 🤔

Reply:
I don't think so, because props and targets exist on the project level, so they wouldn't know about each other.

Reply:
Project-level build in the props file then?

Reply:
I mean then we can't make sure that it only runs once.

Reply:
The scenario I'm explaining is functionally same to referencing NE from each project in the solution, and adding `<MarkdownAnalysis>` to their csproj. This shouldn't cause any issues (also with the stamp file and all).

Reply:
Then only md files inside a project would be analyzed. Excluding every repo root readme for example.

Reply:
We could achieve solution-wide markdown linting by referencing NE from a minimalist project which only references NE and contains the `<MarkdownAnalysis>solution</MarkdownAnalysis>` property. That would be run once per solution.

Since `<MarkdownAnalysis>` is `false` by default, no other project in the solution would run `markdownlint.`

The opposite case would be where you include NE in one or more projects and run `markdownlint` only there.

Reply:
How about using the Web project for this purpose? (If indeed, then document this recommendation.) It's the central project in the solution.

Reply:
That sounds like a good choice!

Reply:
I managed to trick myself into creating a _Lombiq.NodeJs.Extensions.SolutionMarkdownAnalysis_ project because that would be neat. It actually works well as submodule. Then wasted a looooooot of time trying to make it work on NuGet only to realize that this can't and definitely won't work! (because it's indirect use of Lombiq.NodeJs.Extensions so it would require importing the targets and props files, just referencing the package doesn't work) Now excuse me while I go off to sulk in a corner and listen to _And All That Could Have Been_...

Reply:
Maybe the real project was the experience we gathered along the way?

Reply:
Sorry to hear that, @DAud-IcI! Sending some empathy your way 📨!

What do you, @Piedone, mean by:
> Maybe the real project was the experience we gathered along the way?

Reply:
There's this joke, "Maybe the real X was the friends we made along the way." when something doesn't work out.

Reply:
Yeah, it sounded familiar 😃 I just didn't quite get which "real project" you were referring to. But no need to explain further 😉

Reply:
> Additionally, set [these two properties](https://github.com/Lombiq/NodeJs-Extensions/blob/dev/Lombiq.NodeJs.Extensions/Lombiq.NodeJs.Extensions.csproj#L62) in the Web project file to avoid running the JS/SCSS/Assets pipelines for that project:

How come these aren't conditional and you have to opt-out?

Reply:
They come from NPM Targets, and its basically the flesh of it! I'm grateful that @barthamark thought of making them configurable, because that saved my day when building NE and trying to **only forward** the implemented MSBuild integration to consumer projects without NE consuming it itself 😁

Consider this a low-level hack.